diff --git a/libmamba/src/core/repo.cpp b/libmamba/src/core/repo.cpp
index 76adee68..6773ff87 100644
--- a/libmamba/src/core/repo.cpp
+++ b/libmamba/src/core/repo.cpp
@@ -9,6 +9,7 @@
 #include "mamba/core/repo.hpp"
 #include "mamba/core/output.hpp"
 #include "mamba/core/package_info.hpp"
+#include "mamba/core/channel_builder.hpp"
 
 extern "C"
 {
@@ -53,6 +54,7 @@ namespace mamba
         m_repo = repo_create(pool, name.c_str());
         m_repo->appdata = this;
         read_file(index);
+        p_channel = &ChannelBuilder::make_cached_channel(url);
     }
 
     MRepo::MRepo(MPool& pool,
